src/main/java/org/folio/inventory/resources/MoveApi.java

```diff
@@ -0,0 +1,134 @@
+import org.folio.inventory.storage.external.CqlQuery;
+import org.folio.inventory.storage.external.MultipleRecordsFetchClient;
+import org.folio.inventory.support.ItemUtil;
+import org.folio.inventory.support.http.client.OkapiHttpClient;
+import org.folio.inventory.support.http.server.JsonResponse;
+import org.folio.inventory.support.http.server.ServerErrorResponse;
+import org.folio.inventory.support.http.server.ValidationError;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class MoveApi extends AbstractInventoryResource {
+
+  public static final String TO_HOLDINGS_RECORD_ID = "toHoldingsRecordId";
+  public static final String ITEM_IDS = "itemIds";
+
+  public MoveApi(final Storage storage, final HttpClient client) {
+    super(storage, client);
+  }
+
+  @Override
+  public void register(Router router) {
+    router.post("/inventory/items/move")
+      .handler(this::moveItems);
+  }
+
+  private void moveItems(RoutingContext routingContext) {
+
+    WebContext context = new WebContext(routingContext);
+    JsonObject itemsMoveJsonRequest = routingContext.getBodyAsJson();
+
+    Optional<ValidationError> validationError = itemsMoveHasRequiredFields(itemsMoveJsonRequest);
+    if (validationError.isPresent()) {
+      unprocessableEntity(routingContext.response(), validationError.get());
+    } else {
+      String toHoldingsRecordId = itemsMoveJsonRequest.getString(TO_HOLDINGS_RECORD_ID);
+      List<String> itemIdsToUpdate = toListOfStrings(itemsMoveJsonRequest.getJsonArray(ITEM_IDS));
+      try {
+
+        OkapiHttpClient okapiClient = createHttpClient(routingContext, context);
+        CollectionResourceClient itemsStorageClient = createItemsStorageClient(okapiClient, context);
+        MultipleRecordsFetchClient multipleRecordsFetchClient = createItemsFetchClient(itemsStorageClient);
+
+        multipleRecordsFetchClient.find(itemIdsToUpdate, this::fetchItemsCql)
+          .thenAccept(jsons -> {
+            List<Item> itemsToUpdate = jsons.stream()
+              .map(ItemUtil::jsonToServerItem)
+              .map(item -> item.withHoldingId(toHoldingsRecordId))
+              .collect(toList());
+            updateItems(routingContext, context, toHoldingsRecordId, itemIdsToUpdate, itemsToUpdate);
+          });
+      } catch (Exception e) {
+        ServerErrorResponse.internalError(routingContext.response(), e);
+      }
+    }
+  }
+
+  private void updateItems(RoutingContext routingContext, WebContext context, String toHoldingsRecordId,
+      List<String> itemIdsToUpdate, List<Item> itemsToUpdate) {
+    ItemCollection storageItemCollection = storage.getItemCollection(context);
+
+    List<CompletableFuture<Item>> updates = itemsToUpdate.stream()
+      .map(item -> storageItemCollection.update(item.withHoldingId(toHoldingsRecordId)))
```

Comment:
It seems the items should already have the new holdings id, it is populated in the calling method.